packages/big-design/src/components/Counter/spec.tsx

```diff
@@ -1,320 +1,335 @@
       This is a not a custom error component
-      <a href="#" data-testid="test">
-        has a url
-      </a>
+      <a href="#">has a url</a>
     </div>
   );
 
-  const { queryByTestId } = render(<FormGroup>{counterMock({ error: NotAnError, ...requiredAttributes })}</FormGroup>);
+  render(<FormGroup>{<CounterMock value={5} error={NotAnError} />}</FormGroup>);
 
-  expect(queryByTestId('test')).not.toBeInTheDocument();
+  expect(screen.queryByRole('link')).not.toBeInTheDocument();
 });
 
 test('renders both the add and subtract icons', () => {
-  const { getAllByRole } = render(counterMock(requiredAttributes));
+  render(<CounterMock value={5} />);
 
-  const buttons = getAllByRole('button');
+  const buttons = screen.getAllByRole('button');
 
-  expect(buttons.length).toBe(2);
+  expect(buttons).toHaveLength(2);
 });
 
 test('buttons are disabled when value hits max or min', () => {
-  const { getAllByRole, rerender } = render(counterMock({ ...requiredAttributes, value: 0 }));
+  const { rerender } = render(<CounterMock value={0} />);
 
-  const buttons = getAllByRole('button');
+  const buttons = screen.getAllByRole('button');
 
-  expect(buttons[0]).toHaveProperty('disabled', true);
-  expect(buttons[1]).toHaveProperty('disabled', false);
+  expect(buttons[0]).toBeDisabled;
+  expect(buttons[1]).toBeEnabled;
 
-  rerender(counterMock({ ...requiredAttributes, value: 10 }));
+  rerender(<CounterMock value={10} />);
 
-  expect(buttons[0]).toHaveProperty('disabled', false);
-  expect(buttons[1]).toHaveProperty('disabled', true);
+  expect(buttons[0]).toBeEnabled;
+  expect(buttons[1]).toBeDisabled;
 });
 
 test('value prop only accepts whole numbers', () => {
-  const { container } = render(counterMock({ max: 20, ...requiredAttributes }));
+  render(<CounterMock value={5} />);
+
+  const input = screen.getByRole<HTMLInputElement>('textbox');
+  const fractionalValue = '1.7';
 
-  const input = container.getElementsByTagName('input');
-  fireEvent.change(input[0], { target: { value: 1.5 } });
+  userEvent.clear(input);
+  userEvent.type(input, fractionalValue);
 
-  expect(handleChange).toHaveBeenCalledWith(2);
+  expect(input).toHaveValue('17');
 });
 
 test('value increases when increase or decrease icons are clicked', () => {
-  const { getByDisplayValue, container } = render(counterMock(requiredAttributes));
+  const handleChange = jest.fn((num: number) => num);
+  render(<CounterMock value={5} onCountChange={handleChange} />);
 
-  const counter = getByDisplayValue('5') as HTMLInputElement;
-
-  const icons = container.getElementsByTagName('svg');
+  const counterValue = screen.getByRole<HTMLInputElement>('textbox').value;
+  const icons = Array.from(screen.getAllByTitle(/[a-z]{2}crease count/i)).map(
+    (title: HTMLElement): HTMLElement | null => title.parentElement,
+  );
 
-  expect(counter.value).toEqual('5');
-  fireEvent.click(icons[1]);
+  expect(counterValue).toEqual('5');
+  icons[1] && userEvent.click(icons[1]);
```

Comment:
`if`s are not recommended, in other repos we've created a helper that throws when `undefined` or `null`.

```ts
export function assertDefined<T>(a: T | undefined | null): asserts a is T {
  if (a === undefined || a === null) {
    throw new Error('Element not found');
  }
}
```

and then in your tests, you can just use them:

```ts

const [decrease, increase] = Array.from(screen.getAllByTitle(/[a-z]{2}crease count/i)).map(
  (title) => title.parentElement,
);

assertDefined(increase);
assertDefined(decrease);

userEvent.click(increase);

expect(handleChange).toHaveBeenCalledWith(6);

userEvent.click(decrease);
expect(handleChange).toHaveBeenCalledWith(4);
```

A much better alternative will be to just get each button independently:

```ts
const increaseButton = screen.getByRole('button', { name: /increase count/i });
const decreaseButton = screen.getByRole('button', { name: /decrease count/i });
```

Reply:
Thanks about fn `assertDefine`. I will keep it in mind. but in this case your suggestion is much better. changed)